thunder/executors/transformer_engineex.py

```diff
@@ -322,15 +333,7 @@ def _te_functional_linear_meta(
 
             # It's not critical to model the exact shape and dtype of
             # saved_tensors since they are not used in Thunder's meta functions.
-
-            # https://github.com/NVIDIA/TransformerEngine/blob/f2b09d2a206ebaddbec6aa7da1158a449f0457f9/transformer_engine/pytorch/module/linear.py/#L270-L274
-            saved_tensors = (
-                TensorProxy(like=a, shape=a.shape, dtype=float8_e4m3fn),  # saved_inputmat
-                TensorProxy(like=w, shape=w.shape, dtype=uint8),  # fp8_weight (rowwise)
-                TensorProxy(like=w, shape=w.shape, dtype=uint8),  # fp8_weight (colwise)
-                TensorProxy(like=w, shape=w.shape),  # weight
-                None if bias is None else TensorProxy(like=bias, shape=bias.shape),  # bias
-            )
+            saved_tensors = tuple(TensorProxy(like=a, shape=a.shape) for _ in range(_get_num_saved_tensors()))
```

Comment:
It seems that shapes of saved_tensors could be different before/after this PR.
Why not making use of shapes of `o.grad_fn.saved_tensors`?